app/src/main/java/com/example/myapplication/LoginActivity.java

```diff
@@ -49,54 +50,48 @@ public void onStart() {
     public void login(View view) {
 
         // todo error handling
-        final String username = nameText.getText().toString();
         final String emailAddress = emailText.getText().toString();
         final String password = passwordText.getText().toString();
-
+        final LoginActivity thisActivity = this;
         firebaseAuth.signInWithEmailAndPassword(emailAddress, password).addOnCompleteListener(this,
                 new OnCompleteListener<AuthResult>() {
                     @Override
                     public void onComplete(@NonNull Task<AuthResult> task) {
                         if (task.isSuccessful()) {
 
                             FirebaseUser user = firebaseAuth.getCurrentUser();
-                            Toast.makeText(LoginActivity.this, "user logged in",
+                            Toast.makeText(LoginActivity.this, "Login successful!",
                                     Toast.LENGTH_SHORT).show();
-                        }
-                        else {
-
-                            register(username, emailAddress, password);
-                        }
-                    }
-                });
-    }
+                            Handler handler = new Handler();
```

Comment:
What does this handler handling?
I believe that a more informative name is needed here.